### ISSUE TITLE
iOS Simulator geolocation incorrectly stripping negative longitude/latitude

### ISSUE THREAD

**Issue by enriquegh** [appium/appium#15613]
## The problem

When setting the location with AppleScript (i.e. no IDB nor Lyft's set-simulator-location installed) the incorrect coordinates are being passed.
 

## Environment

* Appium version (or git revision) that exhibits the issue: Tried on 1.21.0 and 1.19.1 but all versions are probably affected
* Last Appium version that did not exhibit the issue (if applicable): N/A
* Desktop OS/version used to run Appium: macOS 11.4
* Node.js version (unless using Appium.app|exe): v14.16.1
* Npm or Yarn package manager: npm 6.14.12
* Mobile platform/version under test: iOS 14.5
* Real device or emulator/simulator: Simulator
* Appium CLI or Appium.app|exe: Appium.exe

## Details

Looks like the `-` on the location is getting replaced to a `.` or whatever decimal separator is specified
So something like:
```
[HTTP] --> POST /wd/hub/session/6f40c866-c20e-4c84-ac80-7cba29ae0a2c/location
[HTTP] {"location":{"latitude":"37.79","longitude":"-122.401","altitude":"19.31"}}
[debug] [W3C (6f40c866)] Calling AppiumDriver.setGeoLocation() with args: [{"latitude":"37.79","longitude":"-122.401","altitude":"19.31"},"6f40c866-c20e-4c84-ac80-7cba29ae0a2c"]
```
Longitude is parsed to `.122.401` which turns into `.122`

I traced the logic to here:
https://github.com/appium/appium-ios-simulator/blob/01377f751693197175622d65f2ed4063e8b32dbc/lib/geolocation.js

If I read that right it looks like we are changing all non-digits to the machine's determined decimal separator.
My machine's set in the US so that's `.`.

I think we should do something like `any non-digit nor the minus sign`.
Tried doing this myself but couldn't figure it out.

## Link to Appium logs

https://gist.github.com/enriquegh/af4a33e870ca711e515e1054fe54c426
Local Appium 1.21.0 logs

Also reproduced on Sauce:
https://app.saucelabs.com/tests/cb44730bc9ff44a9b526b114f38f4445#4

## Code To Reproduce Issue

https://github.com/enriquegh/ios-location-appium
You can download the app here:
https://drive.google.com/file/d/1MMMOaGbivahQfmQr4diesWUHL-G2a0hw/view?usp=sharing

**Comment by mykola-mokhnach** [comment]
https://github.com/appium/appium-ios-simulator/pull/310 should do the trick

**Comment by mykola-mokhnach** [comment]
The patch has been published to appium@beta

**Comment by enriquegh** [comment]
I can confirm this is fixed on `1.22.0-beta.0`
Thanks Nick!
